src/http-api/resources/pin.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const debug = require('debug')
+const log = debug('http-api:pin')
+log.error = debug('http-api:pin:error')
+
+exports = module.exports
+
+exports.ls = (request, reply) => {
+  const ipfs = request.server.app.ipfs
+  const types = ipfs.pin.types
+  const path = request.query.arg
+  const type = request.query.type || types.all
+  const onError = (err) => {
+    log.error(err)
+    return reply({
+      Message: `Failed to list pins: ${err.message}`,
+      Code: 0
+    }).code(500)
+  }
+  ipfs.pin.ls(path, { type })
+    .then((result) => {
+      return reply(result)
+    })
+    .catch((err) => {
+      return onError(err)
+    })
+}
```

Comment:
We default to use callbacks rather then promises everywhere in the code to make things consistent. The Promises API is just for consumers.